PayPalNative/src/main/java/com/braintreepayments/api/PayPalNativeRequest.java

```diff
@@ -0,0 +1,4 @@
+package com.braintreepayments.api;
+
+public interface PayPalNativeRequest {
```

Comment:
Currently we have a base class with subclasses for `PayPalRequest`, is there a need for this to be an interface instead, or can we follow the same pattern here?

Reply:
It depends really. Do we want `PayPalNative` to accept `PayPalRequest`? Really dont know if the request are different if its for web or native to be honest. The way I thought of this was that `PayPalNative` can only accept `PayPalNativeRequest`, but for `PayPalNativeVaultRequest` and `PayPalNativeCheckoutRequest` we need the implementation of `PayPalRequest` base class and subclasses (for now). If I made that `PayPalNative` accepts `PayPalRequest`, then the client could send to our module requests of type `PayPalVaultRequest` and `PayPalCheckoutRequest` (from `PayPal` module), and im not sure if we want that. If we need to differentiate requests between those two modules, then this implementation allows for that. If not, then we can just dispose of any request in `PayPalNative` and use the ones in `PayPal`. Does any of this make sense?

Reply:
Makes sense. We're discussing this internally with iOS as well. It's more about SDK alignment what we do here may be influenced by them and vice versa since we strive to keep the SDKs as similar as possible with the exception of a few edge cases.

Reply:
`v4` was also basically a complete re-write for us. We have an established set of conventions in general. It's hard to say what the final decision will be on the types, but I think it's safe to say we can easily change this as we get closer to GA.

Reply:
Maybe we could use the `PayPalNativeCheckoutRequest` concrete type for now and circle back after native XO is fully integrated?